packages/myst-directives/src/code.ts

```diff
@@ -137,3 +139,53 @@ export const codeBlockDirective: DirectiveSpec = {
     return [container];
   },
 };
+
+export const codeCellDirective: DirectiveSpec = {
+  name: 'code-cell',
+  arg: {
+    type: ParseTypesEnum.string,
+    required: true,
+  },
+  options: {
+    tags: {
+      type: ParseTypesEnum.string,
+    },
+  },
+  body: {
+    type: ParseTypesEnum.string,
+  },
+  run(data, vfile): GenericNode[] {
+    const code: Code = {
+      type: 'code',
+      lang: data.arg as string,
+      executable: true,
+      value: data.body as string,
+    };
+    let tags: string[] | undefined;
+    // TODO: this validation should be done in a different place
+    // For example, specifying that the attribute is YAML,
+    // and providing a custom validation on the option.
```

Comment:
@fwkoch I think picking this up in a validation pass on the directives would be very nice. :)